src/toxcore/onion/onion_client.rs

```diff
@@ -292,6 +294,22 @@ impl OnionClient {
         }
     }
 
+    fn is_redundant_ping(pk: PublicKey, search_pk: PublicKey, request_queue: &RequestQueue<AnnounceRequestData>) -> bool {
+        let check_pks = |data: &AnnounceRequestData| -> bool {
+            if let Some(friend_pk) = data.friend_pk {
+                data.pk == pk && search_pk == friend_pk
+            } else {
+                data.pk == pk
```

Comment:
Here you should check that `search_pk` equals our `dht_pk`, otherwise you can find previous announce request for current search request. Then it can be rewritten to:
```rust
let request_search_pk = if Some(friend_pk) = data.friend_pk {
    friend_pk
} else {
    dht_pk
};
data.pk == pk && search_pk == request_search_pk
```
I think it looks better since here you just check that npde's pk and search pk from new request equals to same pks from old request.

Reply:
done